alica_engine/include/engine/RunnableObject.h

```diff
@@ -46,16 +49,39 @@ class RunnableObject
 
     static constexpr int DEFAULT_MS_INTERVAL = 100;
 
-    enum class Flags : uint8_t
+    enum class TracingType : uint8_t
     {
-        INIT_EXECUTED = 1u,
-        TRACING_ENABLED = 1u << 1,
-        RUN_TRACED = 1u << 2
+        // Use the trace context of the lowest ancestor plan for which tracing is not skipped
+        DEFAULT,
+        // Skip tracing
+        SKIP,
+        // Create a root trace
+        ROOT,
+        // Provide a custom trace context
+        CUSTOM
     };
 
+    // Set the tracing type for this runnable object. customTraceContextGetter is required for custom tracing
+    // & this method will be called to get the parent trace context before initialiseParameters is called
+    void setTracing(TracingType type, std::function<std::optional<std::string>()> customTraceContextGetter = {})
+    {
+        _tracingType = type;
+        _customTraceContextGetter = std::move(customTraceContextGetter);
+        if (_tracingType == TracingType::CUSTOM && !_customTraceContextGetter) {
+            ALICA_ERROR_MSG("Custom tracing type specified, but no getter for the trace context is provided. Switching to default tracing type instead");
+            _tracingType = TracingType::DEFAULT;
+        }
+    }
+
     AlicaEngine* _engine;
     const Configuration* _configuration;
+    TracingType _tracingType;
+    std::function<std::optional<std::string>()> _customTraceContextGetter;
     std::unique_ptr<IAlicaTrace> _trace;
+    // True if the behaviour/plan's run method is traced. This ensures we trace the run call once
```

Comment:
```suggestion
    // True if the behaviour/plan's run method has already been logged in the trace
```
similalry for runTraced